File: Redes2/fibonacci_optimized.java
import java.util.*;

class fibonacci {

    static int fib(int n) {
        double phi = (1 + Math.sqrt(5)) / 2;
        return (int) Math.round(Math.pow(phi, n) / Math.sqrt(5));
    }

    public static void main(String[] args) {
        long startTime = System.nanoTime();

        fib(100000);

        long endTime = System.nanoTime();
        long totalTime = endTime - startTime;
        System.out.println(totalTime);
    }
}

File: Redes2/fibonacci_optimized.c
// C Program to find n'th fibonacci Number
#include <stdio.h>
#include <math.h>
#include <sys/time.h>

int fib(int n)
{
    double phi = (1 + sqrt(5)) / 2;
    return round(pow(phi, n) / sqrt(5));
}

int main()
{
    struct timeval stop, start;
    gettimeofday(&start, NULL);

    fib(100000);

    gettimeofday(&stop, NULL);
    printf("took %d\n", stop.tv_usec - start.tv_usec);

    return 0;
}
